docs: ldap secrets hierarchical paths (#27203)
hashicorp/vault: website/content/docs/secrets/ldap.mdx

olcPPolicyUseLockout: TRUE
```

## Hierarchical paths

> The LDAP secrets engine allows you to define role and set names that contain an
> arbitrary number of forward slashes. A name with forward slashes will define a
> hierarchical path structure. For example, you can configure two static roles
> with the names `org/secure` and `org/platform/dev`:

Reviewer: ```suggestion
The LDAP secrets engine lets you define role and set names that contain an
arbitrary number of forward slashes. Names with forward slashes define
hierarchical path structures.

For example, you can configure two static roles with the names `org/secure` and `org/platform/dev`:
```

Style correction: write in active voice

```shell-session
$ vault write ldap/static-role/org/secure \
    username="user1" \
    rotation_period="1h"
Success! Data written to: ldap/static-role/org/secure

$ vault write ldap/static-role/org/platform/dev \
    username="user2" \
    rotation_period="1h"
Success! Data written to: ldap/static-role/org/platform/dev
```

> A name with a hierarchical path allows you to perform LIST operations to query
> the available roles at a specific path with arbitrary depth. A name that ends
> in the forward slash indicates sub-paths reside under the given path. For
> example, continuing from the previous example, we can list all direct children
> under the `org/` path:

Reviewer: ```suggestion
Names with hierarchical paths let you use the Vault API to query the available
roles at a specific path with arbitrary depth. Names that end with a forward
slash indicate that sub-paths reside under that path.

For example, to list all direct children under the `org/` path:
```

Style correction: write in active voice

```shell-session
$ vault list ldap/static-role/org/
Keys
----
platform/
secure
```

> The `platform/` key ends in a forward slash so we can list its sub-keys:

Reviewer: ```suggestion
The `platform/` key also ends in a forward slash. To list the `platform` sub-keys:
```

```shell-session
$ vault list ldap/static-role/org/platform
Keys
----
dev
```

You can read and rotate credentials using the same role name and the respective

> API's. For example,

Reviewer: ```suggestion
APIs. For example,
```

```shell-session
$ vault read ldap/static-cred/org/platform/dev
Key                    Value
---                    -----
dn                     n/a
last_password          a3sQ6OkmXKt2dtx22kAt36YLkkxLsg4RmhMZCLYCBCbvvv67ILROaOokdCaGPEAE
last_vault_rotation    2024-05-03T16:39:27.174164-05:00
password               ECf7ZoxfDxGuJEYZrzgzTffSIDI4tx5TojBR9wuEGp8bqUXbl4Kr9eAgPjmizcvg
rotation_period        5m
ttl                    4m58s
username               user2
```

```shell-session
$ vault write -f ldap/rotate-role/org/platform/dev
```

> Additionally, name with a hierarchical paths allows you to define policies that
> map 1-1 to the LDAP secrets engine's role or set path because Vault policies
> themselves are path-based. Refer to [Vault policies](/vault/docs/concepts/policies)
> to learn more about Vault and Role-Based Access Control (RBAC).

Reviewer: ```suggestion
Since [Vault policies](/vault/docs/concepts/policies) are also path-based,
hierarchical names also let you define policies that map 1-1 to LDAP secrets
engine roles and set paths.
```

Style correction: write in active voice

> Hierarchical path handling is available for the following APIs:

Reviewer: ```suggestion
The following Vault API endpoints support hierarchical path handling:
```

Style correction: write in active voice
